### ISSUE TITLE
Unable to pull images from insecure private repositories while podman-compose up -d

### ISSUE THREAD

**Issue by Mohamed-Abubucker** [containers/podman-compose#381]
**Describe the bug**

Getting below error while pulling the images from insecure private repo as part of `podman-compose up -d`

```
Trying to pull myprivaterepo.com:7071/webapp.eligibility:latest...
Error: initializing source docker://myprivaterepo.com:7071/webapp.eligibility:latest: pinging container registry myprivaterepo.com:7071: Get "https://myprivaterepo.com:7071/v2/": http: server gave HTTP response to HTTPS client
exit code: 125
```

I already added my private repo in /etc/containers/registries.conf file. I am able to pull other base images from that same private repo while doing `buildah bud .` with Dockerfile. 

**To Reproduce**
1. what is the content of the current working directory (ex. `docker-compose.yml`, `.env`, `Dockerfile`, ...etc.): given in the podman-compose logs section.
2. what is the sequence of commands you typed: *podman-compose up -d*

**Expected behavior**
it should pull those images from private insecure repository as part of `podman-compose up -d`

**Actual behavior**
it gives below error

```
Trying to pull myprivaterepo.com:7071/webapp.eligibility:latest...
Error: initializing source docker://myprivaterepo.com:7071/webapp.eligibility:latest: pinging container registry myprivaterepo.com:7071: Get "https://myprivaterepo.com:7071/v2/": http: server gave HTTP response to HTTPS client
exit code: 125
```

**Output**

```
$ podman-compose version
['podman', '--version', '']
using podman version: 3.4.2
podman-composer version  1.0.2
podman --version 
podman version 3.4.2
exit code: 0

$ podman-compose up
...
['podman', '--version', '']
using podman version: 3.4.2
 ** merged:
 {
  "_dirname": "/mnt/d/Code/webapp.webapp",
  "version": "3",
  "webapp": {
    "webapp.webapp": {
      "image": "myprivaterepo.com:7071/webapp.webapp",
      "build": ".",
      "depends_on": [
        "elasticsearch",
        "webapp.postgres",
        "mock.rabbit",
        "apm",
        "webapp.eligibility",
        "pgbouncer"
      ],
      "environment": {
        "WAIT_HOSTS": "webapp.postgres:5432, elasticsearch:9200, apm:8200, mock.rabbit:5672, webapp.eligibility:8080, pgbouncer:5432",
        "WAIT_HOSTS_TIMEOUT": 300
      },
      "user": "1000",
      "volumes": [
        ".:/code"
      ],
      "ports": [
        "7259:7259"
      ]
    },
    "webapp.eligibility": {
      "image": "myprivaterepo.com:7071/webapp.eligibility",
      "expose": [
        8080
      ],
      "depends_on": [
        "apm",
        "eligibility.postgres",
        "webapp.customer",
        "pgbouncer"
      ],
      "environment": {
        "WAIT_HOSTS": "eligibility.postgres:5432, apm:8200, webapp.customer:8080, pgbouncer:5432",
        "WAIT_HOSTS_TIMEOUT": 180
      }
    },
    "webapp.customer": {
      "image": "myprivaterepo.com:7071/webapp.customer",
      "expose": [
        8080
      ],
      "depends_on": [
        "apm",
        "customer.postgres",
        "mock.rabbit",
        "pgbouncer"
      ],
      "environment": {
        "WAIT_HOSTS": "customer.postgres:5432, apm:8200, mock.rabbit:5672, pgbouncer:5432",
        "WAIT_HOSTS_TIMEOUT": 120
      }
    },
    "elasticsearch": {
      "image": "elasticsearch:2.3.5"
    },
    "webapp.postgres": {
      "image": "postgres:11.2",
      "volumes": [
        ".:/code"
      ],
      "ports": [
        "5432:5432"
      ]
    },
    "eligibility.postgres": {
      "image": "postgres:11.2"
    },
    "customer.postgres": {
      "image": "postgres:11.2"
    },
    "pgbouncer": {
      "image": "edoburu/pgbouncer:1.9.0",
      "environment": {
        "DB_USER": "postgres",
        "DB_PASSWORD": "password",
        "POOL_MODE": "transaction"
      },
      "volumes": [
        "./config/pgbouncer.ini:/etc/pgbouncer/pgbouncer.ini:ro"
      ],
      "expose": [
        5432
      ],
      "depends_on": [
        "eligibility.postgres",
        "customer.postgres",
        "webapp.postgres"
      ]
    },
    "mock.rabbit": {
      "image": "rabbitmq:management",
      "expose": [
        15672,
        5672
      ],
      "command": "bash -c \"echo 'loopback_users = none' > /etc/rabbitmq/rabbitmq.conf; sleep 10; rabbitmq-server;\""
    },
    "apm": {
      "image": "store/elastic/apm-server:6.8.4",
      "depends_on": [
        "elasticsearch"
      ]
    }
  }
}
** excluding:  set()
['podman', 'inspect', '-t', 'image', '-f', '{{.Id}}', 'myprivaterepo.com:7071/webapp.webapp']
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_elasticsearch_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=elasticsearch --net webappwebapp_default --network-alias elasticsearch elasticsearch:2.3.5
Error: error creating container storage: the container name "webappwebapp_elasticsearch_1" is already in use by "0d76712677166315b39dd715f9853666cf0e7c01580a37ba12a56773de382825". You have to remove that container to be able to reuse that name.: that name is already in use
exit code: 125
podman start webappwebapp_elasticsearch_1
exit code: 0
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_webapp.postgres_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=webapp.postgres -v /mnt/d/Code/webapp.webapp:/code --net webappwebapp_default --network-alias webapp.postgres -p 5432:5432 postgres:11.2
Error: error creating container storage: the container name "webappwebapp_webapp.postgres_1" is already in use by "8251317b7217a22d467ca723e6b5e2070a8321e02211435a5c601150ca1a2e6b". You have to remove that container to be able to reuse that name.: that name is already in use
exit code: 125
podman start webappwebapp_webapp.postgres_1
exit code: 0
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_eligibility.postgres_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=eligibility.postgres --net webappwebapp_default --network-alias eligibility.postgres postgres:11.2
Error: error creating container storage: the container name "webappwebapp_eligibility.postgres_1" is already in use by "06a300eb7007a297e2bc0e1772f7ed54300178bd1aefeac1cac6d26877b24132". You have to remove that container to be able to reuse that name.: that name is already in use
exit code: 125
podman start webappwebapp_eligibility.postgres_1
exit code: 0
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_customer.postgres_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=customer.postgres --net webappwebapp_default --network-alias customer.postgres postgres:11.2
Error: error creating container storage: the container name "webappwebapp_customer.postgres_1" is already in use by "56e8a7f8a08ccc3081873907f2f0bb4fde9a2096295f9f6b3dd8f0f654a6ab76". You have to remove that container to be able to reuse that name.: that name is already in use
exit code: 125
podman start webappwebapp_customer.postgres_1
exit code: 0
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_mock.rabbit_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=mock.rabbit --net webappwebapp_default --network-alias mock.rabbit --expose 15672 --expose 5672 rabbitmq:management bash -c echo 'loopback_users = none' > /etc/rabbitmq/rabbitmq.conf; sleep 10; rabbitmq-server;
Error: error creating container storage: the container name "webappwebapp_mock.rabbit_1" is already in use by "960600cd52119c6a78497cd8f8117b5b6e2c382810a5984f4681026973695ddf". You have to remove that container to be able to reuse that name.: that name is already in use
exit code: 125
podman start webappwebapp_mock.rabbit_1
exit code: 0
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_apm_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=apm --net webappwebapp_default --network-alias apm store/elastic/apm-server:6.8.4
Error: error creating container storage: the container name "webappwebapp_apm_1" is already in use by "96cd50170cf384006275a25058a2e7b8d91047cab385ac70f78c92c36a1c9911". You have to remove that container to be able to reuse that name.: that name is already in use
exit code: 125
podman start webappwebapp_apm_1
exit code: 0
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_pgbouncer_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=pgbouncer -e DB_USER=postgres -e DB_PASSWORD=password -e POOL_MODE=transaction -v /mnt/d/Code/webapp.webapp/config/pgbouncer.ini:/etc/pgbouncer/pgbouncer.ini:ro --net webappwebapp_default --network-alias pgbouncer --expose 5432 edoburu/pgbouncer:1.9.0
Error: error creating container storage: the container name "webappwebapp_pgbouncer_1" is already in use by "8e30a7cdc53b47e568460836bc9b3a5a23766f7bb7a3735fdc6a5510282855f1". You have to remove that container to be able to reuse that name.: that name is already in use
exit code: 125
podman start webappwebapp_pgbouncer_1
exit code: 0
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_webapp.customer_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=webapp.customer -e WAIT_HOSTS=customer.postgres:5432, apm:8200, mock.rabbit:5672, pgbouncer:5432 -e WAIT_HOSTS_TIMEOUT=120 --net webappwebapp_default --network-alias webapp.customer --expose 8080 myprivaterepo.com:7071/webapp.customer
Trying to pull myprivaterepo.com:7071/webapp.customer:latest...
Error: initializing source docker://myprivaterepo.com:7071/webapp.customer:latest: pinging container registry myprivaterepo.com:7071: Get "https://myprivaterepo.com:7071/v2/": http: server gave HTTP response to HTTPS client
exit code: 125
podman start webappwebapp_webapp.customer_1
Error: no container with name or ID "webappwebapp_webapp.customer_1" found: no such container
exit code: 125
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_webapp.eligibility_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=webapp.eligibility -e WAIT_HOSTS=eligibility.postgres:5432, apm:8200, webapp.customer:8080, pgbouncer:5432 -e WAIT_HOSTS_TIMEOUT=180 --net webappwebapp_default --network-alias webapp.eligibility --expose 8080 myprivaterepo.com:7071/webapp.eligibility
Trying to pull myprivaterepo.com:7071/webapp.eligibility:latest...
Error: initializing source docker://myprivaterepo.com:7071/webapp.eligibility:latest: pinging container registry myprivaterepo.com:7071: Get "https://myprivaterepo.com:7071/v2/": http: server gave HTTP response to HTTPS client
exit code: 125
podman start webappwebapp_webapp.eligibility_1
Error: no container with name or ID "webappwebapp_webapp.eligibility_1" found: no such container
exit code: 125
podman network exists 'webappwebapp_default' || podman network create 'webappwebapp_default'
['podman', 'network', 'exists', 'webappwebapp_default']
podman run --name=webappwebapp_webapp.webapp_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=webapp.webapp -e WAIT_HOSTS=webapp.postgres:5432, elasticsearch:9200, apm:8200, mock.rabbit:5672, webapp.eligibility:8080, pgbouncer:5432 -e WAIT_HOSTS_TIMEOUT=300 -v /mnt/d/Code/webapp.webapp:/code --net webappwebapp_default --network-alias webapp.webapp -p 7259:7259 -u 1000 myprivaterepo.com:7071/webapp.webapp
Error: error creating container storage: the container name "webappwebapp_webapp.webapp_1" is already in use by "8c2b8a28093a8e98d4f94cd9fbadcd7db7afbacd9b18e9414e88c5c389321693". You have to remove that container to be able to reuse that name.: that name is already in use
exit code: 125
podman start webappwebapp_webapp.webapp_1
webappwebapp_webapp.webapp_1
exit code: 0
```

**Environment:**
 - OS:  WSL2 Ubuntu 20.04
 - podman version: 3.4.2
 - podman compose version: (git hex) 1.0.2 (not sure git hex)

**Additional context**

I Just renamed the actual repo URL and image name for some security reason. other than that, everything is the same.

**Comment by muayyad-alsadi** [comment]
you can see the podman command

```
podman run --name=webappwebapp_elasticsearch_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=elasticsearch --net webappwebapp_default --network-alias elasticsearch elasticsearch:2.3.5
```

and it says

```
Error: error creating container storage: the container name "webappwebapp_elasticsearch_1" is already in use by "0d76712677166315b39dd715f9853666cf0e7c01580a37ba12a56773de382825". You have to remove that container to be able to reuse that name.: that name is already in use
```

you need to do down

it's not about pull images

**Comment by Mohamed-Abubucker** [comment]
Hi @muayyad-alsadi ,

I tried down and then deleting all the local images and containers but still facing the same issue while pulling the image.

**Comment by muayyad-alsadi** [comment]
sure, but the error message would be different
it won't be ` the container name is already in use`

can I have the trace of the read problem, not the side problem,

**Comment by Mohamed-Abubucker** [comment]
I can resolve the container name issue by removing containers and images. but the real problem is 

```
Trying to pull myprivaterepo.com:7071/webapp.customer:latest...
Error: initializing source docker://myprivaterepo.com:7071/webapp.customer:latest: pinging container registry myprivaterepo.com:7071: Get "https://myprivaterepo.com:7071/v2/": http: server gave HTTP response to HTTPS client
exit code: 125
```

**Comment by muayyad-alsadi** [comment]
if you go one line above it you would see the full command

```
podman run --name=webappwebapp_webapp.customer_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webappwebapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webappwebapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.webapp --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=webapp.customer -e WAIT_HOSTS=customer.postgres:5432, apm:8200, mock.rabbit:5672, pgbouncer:5432 -e WAIT_HOSTS_TIMEOUT=120 --net webappwebapp_default --network-alias webapp.customer --expose 8080 myprivaterepo.com:7071/webapp.customer
Trying to pull myprivaterepo.com:7071/webapp.customer:latest...
```

the question is what should I pass to the following command to make it work?

```
podman run --name=webappwebapp_webapp.customer_1 -d #...# myprivaterepo.com:7071/webapp.customer
```

and if you run podman manually without podman-compose will it work?

> I already added my private repo in /etc/containers/registries.conf file. I am able to pull other base images from that same private repo while doing buildah bud . with Dockerfile.

the question is not about buildah, the question is about podman,
podman-compose just call podman

https://github.com/containers/podman

**Comment by Mohamed-Abubucker** [comment]
Hi,

I tried removing all my local images and containers, then tried again.

now the container name issue is gone, but still facing the same issue while pulling the image.

```
podman network exists 'webapp_default' || podman network create 'webapp_default'
['podman', 'network', 'exists', 'webapp_default']
podman run --name=webapp_webapp.eligibility_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=webapp --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=webapp --label com.docker.compose.project.working_dir=/mnt/d/Code/webapp.catalog --label com.docker.compose.project.config_files=docker-compose.yml,docker-compose.override.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=webapp.eligibility -e WAIT_HOSTS=eligibility.postgres:5432, apm:8200, webapp.inmate:8080, pgbouncer:5432 -e WAIT_HOSTS_TIMEOUT=180 --net webapp_default --network-alias webapp.eligibility --expose 8080 myprivaterepo.com:8080/webapp.eligibility
Trying to pull myprivaterepo.com:8080/webapp.eligibility:latest...
Error: initializing source docker://myprivaterepo.com:8080/webapp.eligibility:latest: pinging container registry myprivaterepo.com:8080: Get "https://myprivaterepo.com:8080/v2/": http: server gave HTTP response to HTTPS client
exit code: 125
podman start webapp_webapp.eligibility_1
Error: no container with name or ID "webapp_webapp.eligibility_1" found: no such container
exit code: 125
```
Just for information, I mentioned it's working with buildah, I am aware that podman-compose uses podman not buildah. so that I created the issue here.

my problem is 

1. i have added insecure registries in /etc/containers/registries.conf -> buildah recognizes it.
2. but podman-compose throws an error as it expects HTTPS endpoint, but my private repo is insecure (HTTP only).


my request is if you are not sure about the solution, kindly let others give their suggestions.

**Comment by muayyad-alsadi** [comment]
> i have added insecure registries in /etc/containers/registries.conf -> buildah recognizes it.

if podman does not recognize it file a bug to podman

https://github.com/containers/podman

**Comment by Mohamed-Abubucker** [comment]
FYI,

podman CLI has an option to bypass TLS verification.

![image](https://user-images.githubusercontent.com/26508192/147261708-4555d0dc-9c1b-4e95-9ed1-1c61fed54407.png)

I am able to pull by directly using `podman pull myprivaterepo.com:7071/webapp.customer`

while enforcing --tls-verify
```
abubucker@PC-123:~$ podman pull myprivaterepo.com:7071/webapp.customer --tls-verify
Trying to pull myprivaterepo.com:7071/webapp.customer:latest...
Error: initializing source docker://myprivaterepo.com:7071/webapp.customer:latest: pinging container registry myprivaterepo.com:7071: Get "https://myprivaterepo.com:7071/v2/": http: server gave HTTP response to HTTPS client
```
while skipping --tls-verify, its pulling the image.

```
abubucker@PC-123:~$ podman pull myprivaterepo.com:7071/webapp.customer
Trying to pull myprivaterepo.com:7071/webapp.customer:latest...
Getting image source signatures
Copying blob b800e94e7303 skipped: already exists
Copying blob 2e2bafe8a0f4 skipped: already exists
Copying blob 76b8ef87096f skipped: already exists
Copying blob 84a8c1bd5887 skipped: already exists
Copying blob b53ce1fd2746 skipped: already exists
Copying blob 7a803dc0b40f skipped: already exists
Copying blob 0da9fbf60d48 skipped: already exists
Copying blob 04dccde934cf skipped: already exists
Copying blob 73269890f6fd skipped: already exists
Copying blob 8cfe9d97c100 done
Copying blob 3b6e572231f6 done
Copying blob c3da2ce5280c done
Copying blob 8825821e309e done
Copying blob 6c64f34877cc done
Copying blob 0b425a09ac4c done
Copying config 7fca52cfb6 done
Writing manifest to image destination
Storing signatures
7fca52cfb677271fadd151ea3543866df0592a80a62ce2e70c82e8fe1852460d
```

while running `podman-compose up -d`, I am not directly interacting with podman CLI, it is podman-compose. 

so it's the right place to report, not the podman project.

again, kindly let others to respond.

**Comment by Mohamed-Abubucker** [comment]
Hi @muayyad-alsadi,

Finally, I found a solution for this,

before running `podman-compose up -d`, I should run `podman-compose pull` (inside the project directory).

`podman-compose pull` successfully pulls images from the insecure repository.

**Comment by muayyad-alsadi** [comment]
> while enforcing --tls-verify

you can pass specific podman commands via podman-compose 
using something like

```
$ podman-compose --help 
--podman-pull-args
--podman-run-args
```

that is

```
podman-compose --podman-pull-args="--tls-verify=false" --podman-run-args="--tls-verify=false" up -d
```

> my request is if you are not sure about the solution, kindly let others give their suggestions.

I'm the original author of this software.
I'm just trying to understand your case.

you are welcome.

please follow the following ticket
we might implement a way to pass those podman-specific flags in the yaml
so that you don't have to type much

**Comment by Mohamed-Abubucker** [comment]
Hi @muayyad-alsadi,

Thanks for your awesome work.

our initial conversation was a bit deviating from the actual issue. I apologize if I used any inappropriate words/sentences.

The below command works like a charm.

`podman-compose --podman-pull-args="--tls-verify=false" --podman-run-args="--tls-verify=false" up -d`

I tried it before as well but there were some syntax errors.

if possible, kindly add some examples in the documentation (with exact syntax).

thanks a lot.
